File: admin/lib/src/data/models/store.dart
import '../../data/remote/response/store/store_response.dart';

class Store {
  String? storeId;
  String? storeName;
  String? address1;
  String? address2;
  String? address3;
  String? address4;
  String? openingHour;
  String? closingHour;
  String? latitude;
  String? longitude;
  String? imageUrl;
  String? hotlineNumber;
  String? googleMapUrl;
  String? registrationDate;
  String? lastUpdateDate;

  Store({
    this.storeId,
    this.storeName,
    this.address1,
    this.address2,
    this.address3,
    this.address4,
    this.openingHour,
    this.closingHour,
    this.latitude,
    this.longitude,
    this.imageUrl,
    this.hotlineNumber,
    this.googleMapUrl,
    this.registrationDate,
    this.lastUpdateDate,
  });

  Map<String, dynamic> toJson() {
    return {
      "name": storeName,
      "hotlineNumber": hotlineNumber,
      "imageUrl": imageUrl,
      "address1": address1,
      "address2": address2,
      "address3": address3,
      "address4": address4,
      "openingHour": openingHour,
      "closingHour": closingHour,
    };
  }

  factory Store.fromStoreResponse(StoreResponse storeResponse) {
    return Store(
      storeId: storeResponse.storeId,
      storeName: storeResponse.storeName,
      address1: storeResponse.address1,
      address2: storeResponse.address2,
      address3: storeResponse.address3,
      address4: storeResponse.address4,
      openingHour: storeResponse.openingHour,
      closingHour: storeResponse.closingHour,
      latitude: storeResponse.latitude,
      longitude: storeResponse.longitude,
      imageUrl: storeResponse.imageUrl,
      hotlineNumber: storeResponse.hotlineNumber,
      googleMapUrl: storeResponse.googleMapUrl,
      registrationDate: storeResponse.registrationDate,
      lastUpdateDate: storeResponse.lastUpdateDate,
    );
  }
}

File: server/src/main/java/edu/rmit/highlandmimic/model/Store.java
package edu.rmit.highlandmimic.model;

import lombok.Builder;
import lombok.Data;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.mongodb.core.mapping.Document;

@Data
@Builder(toBuilder = true)
@Document("stores")
public class Store {

    @Id
    private String storeId;

    private String storeName;
    private String address1;
    private String address2;
    private String address3;
    private String address4;

    @Builder.Default
    private String openingHour = "07:00";

    @Builder.Default
    private String closingHour = "23:00";
    private String latitude;
    private String longitude;

    private String imageUrl;

    private String hotlineNumber;

    private String googleMapUrl;

    @CreatedDate
    private String registrationDate;

    @LastModifiedDate
    private String lastUpdateDate;

    public static String toGoogleMapUrl(String lat, String lng) {
        return "https://www.google.com/maps/search/?api=1&query=" + lat + "&" + lng;
    }

}
